src/Microsoft.AspNetCore.Razor.Evolution/Legacy/Tokenizer.cs

```diff
@@ -215,12 +215,17 @@ protected TSymbol EndSymbol(SourceLocation start, TSymbolType type)
                     errors[i] = CurrentErrors[i];
                 }
 
-                sym = CreateSymbol(start, Buffer.ToString(), type, errors);
+                sym = CreateSymbol(start, GetSymbolContent(type), type, errors);
             }
             StartSymbol();
             return sym;
         }
 
+        protected virtual string GetSymbolContent(TSymbolType type)
+        {
+            return Buffer.ToString();
+        }
```

Comment:
After a quick look in `CSharpTokenizer`, I have identified many similar possible optimizations:
- New line (one or two chars)
- White space (optimization only for one char white space)
- '@' (transitions)
- Operators
- Brackets
- Integer (when only one digit, but it is maybe too much, even '0' is frequently used)
If you think that it makes sense, I can create another PR later